File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int count = 0;
        Set<Character> seen = new HashSet<>();
        int left = 0, right = 0;
        while (right < s.length()){
            if(!seen.contains(s.charAt(right))){
                seen.add(s.charAt(right++));
            }
            else{
                seen.remove(s.charAt(left++));
            }
            
            count = Math.max(count, seen.size());
        }
        return count;
    }
}

File: Data Structures & Algorithms/trapping-rain-water/submission-0.py
class Solution:
    def trap(self, height: List[int]) -> int:
        n = len(height)
        left, right = 0, n-1
        leftMax = 0
        rightMax = 0
        count = 0
        while left < right:
            if height[left] < height[right]:
                if height[left] < leftMax:
                    count += leftMax - height[left]
                else:
                    leftMax = height[left]
                left += 1

            else:
                if height[right] <= rightMax:
                    count += rightMax - height[right]
                else:
                    rightMax = height[right]
                right -= 1

        return count 